main.tf

```diff
@@ -7,15 +13,11 @@ locals {
   validate_kms_values = !var.kms_encryption_enabled && (var.existing_kms_instance_guid != null || var.kms_key_crn != null) ? tobool("When passing values for var.existing_kms_instance_guid or/and var.kms_key_crn, you must set var.kms_encryption_enabled to true. Otherwise unset them to use default encryption") : true
   # tflint-ignore: terraform_unused_declarations
   validate_kms_vars = var.kms_encryption_enabled && (var.existing_kms_instance_guid == null || var.kms_key_crn == null) ? tobool("When setting var.kms_encryption_enabled to true, a value must be passed for var.existing_kms_instance_guid and var.kms_key_crn") : true
+  kms_service       = module.crn_parser.service_name
+  kms_account_id    = module.crn_parser.account_id
+  kms_key_id        = module.crn_parser.resource
```

Comment:
again your going to need ternary logic here, as the crn parser module wshould only be called if `kms_key_crn` is not null.
If it is null, just default these locals to `null` too as they wont be used